### PR TITLE
Replaces deprecated tslint with eslint

### PR DESCRIPTION
**Pull request by KaiVolland**
## Description

This replaces the deprecated `tslint` package with `eslint` and its corresponding subpackages and configs.

It includes an upgrade of `typescript` to version 4.

## Pull request type

- [x] Dependency updates
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [x] Build related changes

## Do you introduce a breaking change?

- [x] No

### CODE REVIEW

**Review by hwbllmnn — APPROVED** [review]
Nice, please merge!

**Review by ahennr — APPROVED** [review]

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# don't ever lint node_modules
+node_modules
+# don't lint build output (make sure it's set to your correct build folder name)
+dist
```
Comment (jansule) [inline .eslintignore:4]: We should also add the `browser` folder to this list

File: src/Util/FilterUtil.spec.ts [hunk 2]
```diff
@@ -41,7 +41,7 @@ describe('FilterUtil', () => {
   describe('featureMatchesFilter', () => {
     it('returns true if a feature matches the filter', () => {
       const dummyData = TestUtil.getComplexGsDummyData();
-      let matchingFeature = dummyData.exampleFeatures.features[0];
+      const matchingFeature = dummyData.exampleFeatures.features[0];
```
Comment (jansule) [inline src/Util/FilterUtil.spec.ts:44]: Just out of curiosity: Shouldn't `const` be ok here, as we keep the reference to the same object?

**Review by jansule — COMMENTED** [review]
Very nice work @KaiVolland! Just a minor remarks. Feel free to merge afterwards

File: src/Util/FilterUtil.spec.ts (hunk 2 above)
Comment (KaiVolland) [reply src/Util/FilterUtil.spec.ts:44]: `const` is not only ok but recommended by eslint. Thats why i switched :)

**Pull request merged**